# Generated by Django 5.0.1 on 2024-07-06 17:27

from django.db import migrations, IntegrityError, models

def update_and_cleanup_model_names(apps, schema_editor):
    ModelName = apps.get_model('model_manager', 'ModelName')

    # Delete the 'gpt-4' model
    ModelName.objects.filter(name="gpt-4").delete()

    # Update remaining models and decrement their codes
    models_to_update = [
        {"old_code": 2, "new_code": 1},
        {"old_code": 3, "new_code": 2},
        {"old_code": 4, "new_code": 3},
        {"old_code": 5, "new_code": 4},
    ]

    for model in models_to_update:
        ModelName.objects.filter(code=model["old_code"]).update(code=model["new_code"])

    # Update the provider to 'openai'
    ModelName.objects.filter(provider="OpenAI").update(provider="openai")

class Migration(migrations.Migration):

    dependencies = [
        ('model_manager', '0001_initial'),
        # Add the dependency on the previous migration
        ('model_manager', '0002_create_model_names'),
    ]

    operations = [
        migrations.RunPython(update_and_cleanup_model_names)
    ]